lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -282,10 +282,12 @@ sub complex_query {
     }
     else {
         my %js_settings = map { uc($_) => $args{$_} } qw(iso hdd_1);
-        my $subquery = $schema->resultset("JobSettings")->query_for_settings(\%js_settings);
-        push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
+        if (%js_settings) {
+            my $subquery = $schema->resultset("JobSettings")->query_for_settings(\%js_settings);
+            push(@conds, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}});
+        }
```

Comment:
tons actually. I just moved the if here as I noticed some queries that go like ... where job_id in (select job_id from job_settings) - which sounded a bit harsh to the database to me .